command.go

```diff
@@ -1059,11 +1059,8 @@ func (c *Command) ExecuteC() (cmd *Command, err error) {
 		cmd.commandCalledAs.name = cmd.Name()
 	}
 
-	// We have to pass global context to children command
-	// if context is present on the parent command.
-	if cmd.ctx == nil {
-		cmd.ctx = c.ctx
-	}
+	// Pass context of root command to child command.
+	cmd.ctx = c.ctx
```

Comment:
I'm worried think this could be a breaking change...
If a program calls `ExecuteContext` on another command than that command would get a different context on the root if we always pass it down.

I'll be honest that I don't know if this is a valid scenario.  But we have to be very careful since Cobra is used by so many programs.  I'll have to test to see if there is a potential problem.

Reply:
I understand what you mean. In fact without other changes executing a subcommand with a context would not have effect and it would use the root's context. The line of code that makes that happen turned out to not be covered by tests: https://github.com/spf13/cobra/blob/923592041e4ae38e2ae5c3cc66c3eb1e0c27261f/command.go#L1012

I added a test to cover this scenario and changed the line such that a command's context is also passed down to the root if execute was called on a subcommand.

Reply:
I had a similar pull request: https://github.com/spf13/cobra/issues/2077
We can combine these two to one I think. Thanks.